docs/examples/classification.ipynb

```diff
@@ -0,0 +1,946 @@
+    "from sklearn.model_selection import HalvingGridSearchCV\n",
+    "from sklearn.ensemble import HistGradientBoostingClassifier\n",
+    "\n",
+    "\n",
+    "\n",
+    "col_trans_bt = ColumnTransformer(\n",
+    "    [\n",
+    "        (\"categorical_features\", OrdinalEncoder(), x_discrete),\n",
+    "    ]\n",
+    ")\n",
+    "m_hgbt_base = Pipeline(\n",
+    "    [\n",
+    "        (\"column_transformer\", col_trans_bt),\n",
+    "        (\n",
+    "            \"model\", HistGradientBoostingClassifier(\n",
+    "                loss=\"log_loss\",\n",
+    "                categorical_features=list(range(len(x_discrete))),\n",
+    "                max_iter=200,\n",
+    "                random_state=33,\n",
+    "            )\n",
+    "        ),\n",
+    "    ]\n",
+    ")\n",
+    "\n",
+    "param_grid = {\n",
+    "    \"model__learning_rate\": [0.02, 0.05],\n",
+    "    \"model__l2_regularization\": [0.1, 0.2, 0.3, 1],\n",
+    "    \"model__max_depth\": [None, 1, 2, 3, 5],\n",
+    "    \"model__max_leaf_nodes\": [3, 10, 30, 50]\n",
+    "}\n",
+    "\n",
+    "\n",
+    "# successive halfing grid search (CV) on the training data\n",
+    "shgs = HalvingGridSearchCV(\n",
+    "    m_hgbt_base,\n",
+    "    param_grid=param_grid,\n",
+    "    cv=5,\n",
+    "    scoring=\"neg_log_loss\",\n",
+    "    random_state=321,\n",
+    ").fit(X_train, y_train)\n",
+    "\n",
+    "m_hgbt = shgs.best_estimator_"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "({'model__l2_regularization': 0.1,\n",
+       "  'model__learning_rate': 0.05,\n",
+       "  'model__max_depth': None,\n",
+       "  'model__max_leaf_nodes': 50},\n",
+       " -0.012953509513355315)"
+      ]
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "shgs.best_params_, shgs.best_score_"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "200"
+      ]
+     },
+     "execution_count": 12,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "m_hgbt[-1].n_iter_"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Calibration Assessment\n",
+    "To make the code easier, we put together the predictions of all of our models on the test set in a single dataframe."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def _helper(mod, X): return pd.Series(mod.predict_proba(X)[:, 1])\n",
+    "models = [m_trivial, m_logistic, m_hgbt]\n",
+    "df_pred_test = pd.concat([_helper(el, X_test) for el in models], axis=1, keys=['Trivial', 'Logistic', 'HGBT'])\n"
```

Comment:
```suggestion
    "df_pred_test = pd.concat(\n"
    "    [_helper(m, X_test) for m in models], axis=1, keys=['Trivial', 'Logistic', 'HGBT']\n"
    ")\n"
```

Reply:
I did this part with autopep8, I hope it's ok? 8d3e7206d7ee6c0c4ab1fa004bdf4dbf33e7a346